### DISCUSSION TITLE
Enable default property overrides by the theme

### DISCUSSION THREAD

**Discussion by BennyAlex** [KevinVandy/material-react-table#381] [Ideas]
Hello,
it would be nice if we can define defaults for the props (and therefore also for the style) in the theme file itself.
Currently I import all TableDefaults myself and spread them  {...TableDefaults} into every table.

Mui has the ability to define "defaultProps: { }" for every component which is then used by the theme.

**Comment by KevinVandy** [comment]
MRT already respects the MUI theme. You can style any of the components with your MUI theme already. Here's an [example](https://www.material-react-table.com/docs/guides/customize-components#custom-mui-theme-example)

Otherwise, I'm not quite sure what you mean by `TableDefaults`

**Reply by BennyAlex** [reply]
@KevinVandy 
Hey, TableDefaults is my own props-definition object I pass to every instance of MaterialReactTable, for exaple it has the following props:
export const MaterialReactTableDefaults = {
  enableTopToolbar: false,
  enableColumnActions: false,
  initialState: {
    showColumnFilters: true,
    showGlobalFilter: false,
    pagination: { pageSize: 15 }
  },
}

and then in my template:
<MaterialReactTable
       columns={columns}
      data={rows}
      {...MaterialReactTableDefaults}
 />

So afaik its not possible to define property defaults for MaterialReactTable, or?

https://mui.com/material-ui/customization/theme-components/
![image](https://user-images.githubusercontent.com/17402724/219509090-bed509ef-e59e-4c07-89d3-6c6d2d190eae.png)
